tests/test_lab/test_detectors.py

```diff
@@ -191,62 +195,55 @@ def test_homodyne_on_2mode_squeezed_vacuum(self, s, outcome):
                 [2 * np.sqrt(s * (1 + s)) * outcome / (np.exp(-2 * r) + 1 + 2 * s), 0.0]
-            ) * np.sqrt(2 * hbar)
-            assert np.allclose(remaining_state.means, means)
+            )
+            assert np.allclose(remaining_state.means.numpy(), means)
 
-    @given(
-        s=st.floats(1.0, 10.0), outcome=none_or_(st.floats(-10.0, 10.0)), angle=st.floats(0, np.pi)
-    )
+    @given(s=st.floats(1.0, 10.0), outcome=none_or_(st.floats(-2, 2)), angle=st.floats(0, np.pi))
     def test_homodyne_on_2mode_squeezed_vacuum_with_angle(self, s, outcome, angle):
         """Check that homodyne detection on TMSV works with an arbitrary quadrature angle"""
-        homodyne = Homodyne(quadrature_angle=angle, result=outcome)
-        r = homodyne.r.value
+        r = settings.HOMODYNE_SQUEEZING
+        homodyne = Homodyne(quadrature_angle=angle, result=outcome, r=r)
         remaining_state = TMSV(r=np.arcsinh(np.sqrt(abs(s)))) << homodyne[0]
         denom = 1 + 2 * s * (s + 1) + (2 * s + 1) * np.cosh(2 * r)
-        cov = (
-            hbar
-            / 2
-            * np.array(
+        cov = (hbar / 2) * np.array(
+            [
+                [
+                    1
+                    + 2 * s
+                    - 2
+                    * s
+                    * (s + 1)
+                    * (1 + 2 * s + np.cosh(2 * r) + np.cos(2 * angle) * np.sinh(2 * r))
+                    / denom,
+                    2 * s * (1 + s) * np.sin(2 * angle) * np.sinh(2 * r) / denom,
+                ],
                 [
-                    [
+                    2 * s * (1 + s) * np.sin(2 * angle) * np.sinh(2 * r) / denom,
+                    (
                         1
                         + 2 * s
-                        - 2
-                        * s
-                        * (s + 1)
-                        * (1 + 2 * s + np.cosh(2 * r) + np.cos(2 * angle) * np.sinh(2 * r))
-                        / denom,
-                        2 * s * (1 + s) * np.sin(2 * angle) * np.sinh(2 * r) / denom,
-                    ],
-                    [
-                        2 * s * (1 + s) * np.sin(2 * angle) * np.sinh(2 * r) / denom,
-                        (
-                            1
-                            + 2 * s
-                            + (1 + 2 * s * (1 + s)) * np.cosh(2 * r)
-                            + 2 * s * (s + 1) * np.cos(2 * angle) * np.sinh(2 * r)
-                        )
-                        / denom,
-                    ],
-                ]
-            )
+                        + (1 + 2 * s * (1 + s)) * np.cosh(2 * r)
+                        + 2 * s * (s + 1) * np.cos(2 * angle) * np.sinh(2 * r)
+                    )
+                    / denom,
+                ],
+            ]
         )
-        assert np.allclose(remaining_state.cov, cov)
+        assert np.allclose(remaining_state.cov.numpy(), cov, atol=1e-5)
         # TODO: figure out why this is not working
```

Comment:
still not working?

Reply:
Nope 🫤 my suspects are units however I couldn't get it right.